src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts

```diff
@@ -144,6 +148,59 @@ export class WorkspaceUiSettingsClientWrapper {
       return wrapperOptions.client.update(type, id, attributes, options);
     };
 
+    const deleteWithDefaultIndexPatternCheck = async (
+      type: string,
+      id: string,
+      options: SavedObjectsDeleteOptions = {}
+    ) => {
+      if (type !== 'index-pattern') {
+        return wrapperOptions.client.delete(type, id, options);
+      }
+      const indexPatternObject = await wrapperOptions.client.get<WorkspaceAttribute>(
+        'index-pattern',
+        id
+      );
+      if (!indexPatternObject.workspaces || indexPatternObject.workspaces?.length === 0) {
+        return wrapperOptions.client.delete(type, id, options);
+      }
+      const workspaceObjects = await this.getWorkspaceTypeEnabledClient(
+        wrapperOptions.request
+      ).bulkGet<{ uiSettings?: { [DEFAULT_INDEX_PATTERN_UI_SETTINGS_ID]?: string } }>(
+        indexPatternObject.workspaces.map((workspaceId) => ({
+          type: WORKSPACE_TYPE,
+          id: workspaceId,
+        }))
+      );
+
+      const defaultIndexPatternWorkspaces = workspaceObjects.saved_objects.filter(
+        ({ attributes }) => attributes?.uiSettings?.[DEFAULT_INDEX_PATTERN_UI_SETTINGS_ID] === id
+      );
+      try {
+        await this.getWorkspaceTypeEnabledClient(wrapperOptions.request).bulkUpdate(
+          defaultIndexPatternWorkspaces.map((savedObject) => ({
```

Comment:
can we check defaultIndexPatternWorkspaces is not empty first to reduce unnecessary workspace update? 

also will bulkUpdate an empty list throw errors?

Reply:
The `bulkUpdate` won't throw errors for empty list. But you're right, we can add a empty check logic for it.